packages/headless-react/package.json

```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@coveo/headless-react",
+  "sideEffects": false,
+  "version": "1.0.0",
+  "description": "React specific wrapper for the Headless component library",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/coveo/ui-kit.git",
+    "directory": "packages/headless-react"
+  },
+  "scripts": {
+    "build": "npm run clean && npm run compile",
+    "clean": "rimraf -rf dist",
+    "compile:esm": "tsc -p tsconfig.esm.json",
+    "compile:cjs": "tsc -p tsconfig.cjs.json",
+    "compile:iife": "rollup --config rollup.config.js",
+    "compile": "concurrently \"npm run compile:esm\" \"npm run compile:cjs\"  \"npm run compile:iife\"",
+    "npm:publish": "node ../../scripts/deploy/publish.mjs"
+  },
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/index.esm.js",
+  "types": "./dist/index.d.ts",
+  "files": [
+    "dist/"
+  ],
+  "dependencies": {
+    "@coveo/headless": "2.7.0",
+    "fast-equals": "4.0.3"
+  },
+  "devDependencies": {
+    "@types/react": "18.0.17",
+    "@types/react-dom": "18.0.6",
+    "@types/node": "18.11.18",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "rollup": "^2.79.0",
+    "rollup-plugin-polyfill-node": "^0.10.2",
+    "rollup-plugin-terser": "^7.0.2",
+    "@rollup/plugin-commonjs": "^22.0.2",
+    "@rollup/plugin-node-resolve": "^14.1.0",
+    "@rollup/plugin-typescript": "^8.5.0"
+  },
+  "peerDependencies": {
+    "react": ">=18.0.0",
+    "react-dom": ">=18.0.0"
```

Comment:
I'm hesitant on whether we should only support newer React versions for this package, just like `atomic-react` or support more versions.

I copied this from `atomic-react` for now.